auth/services/oauth/oauth.go

```diff
@@ -239,6 +318,54 @@ func (s *service) validateSubject(context *validationContext) error {
 // validate the legal base, according to RFC003 §5.2.1.7 if sid is present
 // use consent store
 func (s *service) validateAuthorizationCredentials(context validationContext) error {
+	// filter on authorization credentials
+	vcs, err := context.verifiableCredentials()
+	if err != nil {
+		return fmt.Errorf(errInvalidVCClaim, err)
+	}
+	j := 0
+	for _, vc := range vcs {
+		if vc.IsType(*credential.NutsAuthorizationCredentialTypeURI) {
+			vcs[j] = vc
+			j++
+		}
+	}
+
+	vcs = vcs[:j]
+
+	// no auth creds, return
+	if len(vcs) == 0 {
+		return nil
+	}
+
+	iat := context.jwtBearerToken.IssuedAt()
+	iss := context.jwtBearerToken.Issuer()
+	sub := context.jwtBearerToken.Subject()
+
+	for _, authCred := range vcs {
+		// first check if the VC is valid
+		if err := s.vcValidator.Validate(authCred, true, &iat); err != nil {
```

Comment:
In the eOverdracht Leveranciersspecificatie "Retrieve eTransfer" and "Complete Transfer" sections of the sequence diagram, where the receiving care organization retrieves the FHIR compositions and updates the FHIR task it must send a Nuts Auth VC for authorisation, as per spec. It also provides the access token (containing the VC?) which appears to be issued by the sending care organization at step 19. In that case, the issuer of the JWT and VC are the same, so do these checks below apply to those steps as well? (`jwt.issuer == vc.subject.id && jwt.subject == vc.issuer`)

https://nuts-foundation.gitbook.io/bolts/eoverdracht/leveranciersspecificatie

Reply:
No the issuer of the JWT grant token is the actor.